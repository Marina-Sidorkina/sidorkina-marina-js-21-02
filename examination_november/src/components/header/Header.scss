.header {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: 2000px;
  min-width: 320px;
  height: 60px;
  background-color: #ffffff;

  &__container {
    box-sizing: border-box;
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    max-width: 1200px;
    min-width: 320px;
    height: 60px;
    padding: 0 15px;
  }
}
