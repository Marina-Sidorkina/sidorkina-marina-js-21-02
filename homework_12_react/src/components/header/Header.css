.header {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  width: 1024px;
  height: 75px;
  padding: 15px;
  border-bottom: 5px solid darkmagenta;
}

.header__logo {
  margin-right: 35px;
  padding-top: 4px;
  font-weight: bold;
  font-size: 18px;
  color: darkmagenta;
}

.header__title {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 28px;
}
