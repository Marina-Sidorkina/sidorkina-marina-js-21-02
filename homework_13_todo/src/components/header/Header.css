.header {
  box-sizing: border-box;
  width: 720px;
  height: 75px;
  padding-left: 15px;
  border-bottom: 3px solid indigo;
}

.header__title {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 32px;
  line-height: 75px;
}
