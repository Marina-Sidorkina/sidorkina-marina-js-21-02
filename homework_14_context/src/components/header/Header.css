.header__title {
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 20px;
  font-size: 52px;
}
