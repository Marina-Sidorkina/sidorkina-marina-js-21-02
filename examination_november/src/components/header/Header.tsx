import React from 'react';
import './Header.scss';
import Logo from '../logo/Logo';

const Header = () => (
  <header className="header">
    <div className="header__container">
      <Logo />
    </div>
  </header>
);

export default Header;
