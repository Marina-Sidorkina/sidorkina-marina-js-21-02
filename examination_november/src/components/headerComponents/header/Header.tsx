import React, { useContext } from 'react';
import './Header.scss';
import Logo from '../logo/Logo';
import Navigation from '../navigation/Navigation';
import Authorization from '../authorization/Authorization';
import { ThemeContext } from '../../../contexts/ThemeContext';

const Header = () => {
  const themeContext = useContext(ThemeContext);

  return (
    <header className={`${themeContext.darkTheme ? 'header header_dark' : 'header'}`}>
      <div className="header__container">
        <Logo />
        <Navigation />
        <Authorization />
      </div>
    </header>
  );
};

export default Header;
