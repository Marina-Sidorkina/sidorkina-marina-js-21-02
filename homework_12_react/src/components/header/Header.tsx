import React from "react";
import "./Header.css";

class Header extends React.Component {
  render() {
    return (
      <header className="header">
        <div className="header__logo">PIANOS</div>
        <h1 className="header__title">Пианино и рояли</h1>
      </header>
    )
  }
}

export default Header;
