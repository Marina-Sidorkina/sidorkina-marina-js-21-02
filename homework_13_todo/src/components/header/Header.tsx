import React from "react";
import "./Header.css";

class Header extends React.Component {
  render() {
    return (
      <header className="header app__header">
        <h1 className="header__title">TodoList</h1>
      </header>
    );
  }
}

export default Header;
