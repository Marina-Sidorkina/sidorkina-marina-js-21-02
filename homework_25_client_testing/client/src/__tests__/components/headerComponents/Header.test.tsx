import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import Header from '../../../components/headerComponents/header/Header';

configure({ adapter: new Adapter() });

describe('Header component test', () => {
  test('should render Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.length).toEqual(1);
  });

  test('should render Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
