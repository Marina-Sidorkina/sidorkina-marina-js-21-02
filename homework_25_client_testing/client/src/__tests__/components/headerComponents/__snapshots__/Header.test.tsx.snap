// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test should render Header 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <Logo />
    <Navigation />
    <Authorization />
  </div>
</header>
`;
